### PR TITLE
Add call to inquiry submission API

### PR DESCRIPTION
**Pull request by pviotti**
I had to cast to `unknown` the event fired in the form due to some issues with react-bootstrap typing (see, for instance, https://github.com/DefinitelyTyped/DefinitelyTyped/issues/16208 ). If you know a cleaner way of doing this, let me know.

### CODE REVIEW

File: src/PetCard.tsx [hunk 1]
```diff
@@ -33,7 +44,44 @@ export class PetCard extends React.Component<PetCardProps, PetCardState> {
         this.setState({ showModal: true });
     }
 
+    submitInquiry(event: any) {
+        event.preventDefault();
+        fetch(`https://codess-shelter.azurewebsites.net/api/v1/inquiries`, {
+            method: "POST",
+            mode: "cors",
+            cache: "no-cache", 
+            credentials: "same-origin",
```
Comment (notheotherben) [inline src/PetCard.tsx]: This API endpoint doesn't require auth, so you can set `credentials: "omit"` to be a bit more cautious about when we send those over the wire.

File: src/PetCard.tsx [hunk 2]
```diff
@@ -33,7 +44,44 @@ export class PetCard extends React.Component<PetCardProps, PetCardState> {
...
+            headers: {
+                "Content-Type": "application/json",
+            },
+            redirect: "follow",
+            referrer: "no-referrer",
+            body: JSON.stringify(
+                {
```
Comment (notheotherben) [inline src/PetCard.tsx]: Would it make sense to cast this to an `InquiryV1` to avail of TypeScript's type checking?

File: src/PetCard.tsx [hunk 3]
```diff
@@ -33,7 +44,44 @@ export class PetCard extends React.Component<PetCardProps, PetCardState> {
...
+                    petId: this.state.pet.id,
+                    name: this.state.name,
+                    email: this.state.email,
+                    message: this.state.message
+                }), 
+        }).then(response => {
+            if (response.status !== 201) {
+                console.log(response.json())
```
Comment (notheotherben) [inline src/PetCard.tsx]: Not sure if you want to include logging here? 🤔 (maybe just a miss on commenting it out?)

File: src/PetCard.tsx [hunk 4]
```diff
@@ -33,7 +44,44 @@ export class PetCard extends React.Component<PetCardProps, PetCardState> {
...
+                this.setState({ formState: "error" });
+                throw new Error(`HTTP Error ${response.statusText}`);
+            }
+            return response.json();
+        }
+        ).then(data => {
+            //console.log(data);
+            this.setState({
+                inquiryId: (data as any).id,
```
Comment (notheotherben) [inline src/PetCard.tsx]: Maybe we want to use `(data as InquiryV1).id` or, even better, replace `.then(data => ...` with `.then((data: InquiryV1) => ...` to provide the type information to the whole block. What do you think?

**Review by notheotherben — COMMENTED** [review]
Looking good 👍

File: src/PetCard.tsx (hunk 1 above)
Comment (pviotti) [reply src/PetCard.tsx]: ok, done

File: src/PetCard.tsx (hunk 2 above)
Comment (pviotti) [reply src/PetCard.tsx]: make sense. Added InquiryV1 interface and improved typing of calls payload. Tbh though, due to some buggy typing in react-bootstrap, Typescript has been mostly a hindrance in this little project. That's why I think we should leave the instructions to be Javascript or JS/Ts agnostic

**Review by notheotherben — APPROVED** [review]
Looks awesome! 👍

**Pull request merged**